provider/kv/kv.go

```diff
@@ -12,45 +11,47 @@ import (
 	"github.com/cenk/backoff"
 	"github.com/containous/traefik/job"
 	"github.com/containous/traefik/log"
+	"github.com/containous/traefik/provider"
 	"github.com/containous/traefik/safe"
 	"github.com/containous/traefik/types"
 	"github.com/docker/libkv"
 	"github.com/docker/libkv/store"
 )
 
-// Kv holds common configurations of key-value providers.
-type Kv struct {
-	BaseProvider `mapstructure:",squash"`
-	Endpoint     string     `description:"Comma sepparated server endpoints"`
-	Prefix       string     `description:"Prefix used for KV store"`
-	TLS          *ClientTLS `description:"Enable TLS support"`
-	storeType    store.Backend
-	kvclient     store.Store
+// Provider holds common configurations of key-value providers.
+type Provider struct {
+	provider.BaseProvider `mapstructure:",squash"`
+	Endpoint              string              `description:"Comma sepparated server endpoints"`
+	Prefix                string              `description:"Prefix used for KV store"`
+	TLS                   *provider.ClientTLS `description:"Enable TLS support"`
+	StoreType             store.Backend
+	Kvclient              store.Store
```

Comment:
I'm not sure that's a good idea to have these fields exported. They will be added automatically to the flags traefik --help (to be verified).